src/contexts/products-context.js

```diff
@@ -0,0 +1,47 @@
+import { useReducer, useContext, createContext } from "react";
+
+const ProductContext = createContext();
+
+function reducerFunc(prevState, action) {
+  switch (action.TYPE) {
+    case "SHOW_ONLY_IN_STOCK": {
+      return {
+        ...prevState,
+        showOnlyInStock: !prevState.showOnlyInStock,
+      };
+    }
+
+    case "SHOW_ONLY_FAST_DELIVERY": {
+      return {
+        ...prevState,
+        showOnlyFastDelivery: !prevState.showOnlyFastDelivery,
+      };
+    }
```

Comment:
I suggest the same as other reviewer that you move all the reducer functions from context to a folder named reducer consisting of all the reducer functions at one place.